tests/test_nmapconnector.py

```diff
@@ -0,0 +1,254 @@
+    )
+
+    assert (
+        nm_v6.get_results()
+        .get(NmapTestDefaultValues.HOST6)
+        .get("tcp")
+        .get(NmapTestDefaultValues.PORT6, False)
+    )
+
+
+def test_nmapconnector_scan_error() -> None:
+    """
+    Raise NmapConnectorScanError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.scan",
+        side_effect=NmapConnectorScanError("test"),
+    ):
+        with raises(NmapConnectorScanError) as scan_error:
+            nm.scan(host=NmapTestDefaultValues.HOST4)
+        assert "Error occured in Nmap Connector module: test" == str(scan_error.value)
+
+
+def test_nmpaconnector_scan_without_any_args() -> None:
+    """
+    Test NmapConnector scan running without any args
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host="", arguments="", ports="", sudo=False)
+    assert "'' does not appear to be an IPv4 or IPv6 address" in str(scan_error.value)
+
+
+def test_nmpaconnector_scan_bad_argument() -> None:
+    """
+    Test NmapConnector scan running with bad arguments
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host=NmapTestDefaultValues.HOST4, arguments="--bad-argument")
+    assert "nmap: unrecognized option" in str(scan_error.value) \
+           and "--bad-argument" in str(scan_error.value)
+
+
+def test_nmapconnector_get_results_ipv4() -> None:
+    """
+    Check if we can successfully get NmapConnector
+    scan results for IPv4
+    :return:
+    """
+    assert nm_v4.get_results()
+
+
+def test_nmapconnector_get_results_ipv6() -> None:
+    """
+    Check if we can successfully get NmapConnector
+    scan results for IPv6
+    :return:
+    """
+    assert nm_v6.get_results()
+
+
+def test_nmapconnector_get_results_with_default_values() -> None:
+    """
+    Check if we can successfully get scan results
+    from NmapConnector with default values
+    :return:
+    """
+    assert empty_nm.get_results() == {}
+
+
+def test_nmapconnector_get_results_error() -> None:
+    """
+    Raise NmapConnectorGetResultsError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.get_results",
+        side_effect=NmapConnectorGetResultsError("test"),
+    ):
+        with raises(NmapConnectorGetResultsError) as get_results_error:
+            nm.get_results()
+        assert "Error occured in Nmap Connector module: test" == str(
+            get_results_error.value
+        )
+
+
+def test_nmapconnector_get_results_count_ipv4() -> None:
+    """
+    Check if we can successfully get the count of
+    NmapConnector scan results for IPv4
+    :return:
+    """
+    nm_v4.get_results_count()
+
+
+def test_nmapconnector_get_results_count_ipv6() -> None:
+    """
+    Check if we can successfully get the count of
+    NmapConnector scan results for IPv6
+    :return:
+    """
+    nm_v6.get_results_count()
```

Comment:
Maybe we should use `assert` here like on lines 186, 195? We need to check that value is not null or not empty.